aiida/manage/configuration/migrations/migrations.py

```diff
@@ -8,136 +8,282 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """Define the current configuration version and migrations."""
+from typing import Any, Dict, Protocol
 
-__all__ = ('CURRENT_CONFIG_VERSION', 'OLDEST_COMPATIBLE_CONFIG_VERSION')
+from aiida.common import exceptions
+
+__all__ = (
+    'CURRENT_CONFIG_VERSION', 'OLDEST_COMPATIBLE_CONFIG_VERSION', 'get_current_version', 'check_and_migrate_config',
+    'config_needs_migrating', 'upgrade_config', 'downgrade_config'
+)
+
+ConfigType = Dict[str, Any]
 
 # The expected version of the configuration file and the oldest backwards compatible configuration version.
 # If the configuration file format is changed, the current version number should be upped and a migration added.
 # When the configuration file format is changed in a backwards-incompatible way, the oldest compatible version should
 # be set to the new current version.
+
 CURRENT_CONFIG_VERSION = 5
 OLDEST_COMPATIBLE_CONFIG_VERSION = 5
 
 
-class ConfigMigration:
-    """Defines a config migration."""
+class SingleMigration(Protocol):
+    """A single migration of the configuration."""
+    initial: int
```

Comment:
I take it these `initial` and `final` are like the `down_revision` and `revision` in the SQLA migrations? Would it make sense to use the same terminology here?

Reply:
They are indeed, will do